src/openforms/contrib/objects_api/setup_configuration/tests/test_objects_api_config.py

```diff
@@ -0,0 +1,224 @@
+
+        self.assertEqual(config.drc_service, None)
+        self.assertEqual(config.catalogi_service, None)
+        self.assertEqual(config.catalogue_domain, "")
+        self.assertEqual(config.catalogue_rsin, "")
+        self.assertEqual(config.organisatie_rsin, "")
+        self.assertEqual(config.iot_submission_report, "")
+        self.assertEqual(config.iot_submission_csv, "")
+        self.assertEqual(config.iot_attachment, "")
+
+    def test_execute_with_all_fields(self):
+        execute_single_step(
+            ObjectsAPIConfigurationStep, yaml_source=CONFIG_FILE_PATH_ALL_FIELDS
+        )
+
+        self.assertEqual(ObjectsAPIGroupConfig.objects.count(), 1)
+
+        config = ObjectsAPIGroupConfig.objects.get()
+
+        self.assertEqual(config.name, "Config 1")
+        self.assertEqual(config.identifier, "config-1")
+        self.assertEqual(config.objects_service, self.objects_service)
+        self.assertEqual(config.objecttypes_service, self.objecttypes_service)
+        self.assertEqual(config.drc_service, self.drc_service)
+        self.assertEqual(config.catalogi_service, self.catalogi_service)
+        self.assertEqual(config.catalogue_domain, "TEST")
+        self.assertEqual(config.catalogue_rsin, "000000000")
+        self.assertEqual(config.organisatie_rsin, "000000000")
+        self.assertEqual(config.iot_submission_report, "PDF Informatieobjecttype")
+        self.assertEqual(config.iot_submission_csv, "CSV Informatieobjecttype")
+        self.assertEqual(config.iot_attachment, "Attachment Informatieobjecttype")
+
+    def test_execute_is_idempotent(self):
+        def make_assertions():
+            self.assertEqual(ObjectsAPIGroupConfig.objects.count(), 1)
+
+            config = ObjectsAPIGroupConfig.objects.get()
+
+            self.assertEqual(config.name, "Config 1")
+            self.assertEqual(config.identifier, "config-1")
+            self.assertEqual(config.objects_service, self.objects_service)
+            self.assertEqual(config.objecttypes_service, self.objecttypes_service)
+            self.assertEqual(config.drc_service, self.drc_service)
+            self.assertEqual(config.catalogi_service, self.catalogi_service)
+            self.assertEqual(config.catalogue_domain, "TEST")
+            self.assertEqual(config.catalogue_rsin, "000000000")
+            self.assertEqual(config.organisatie_rsin, "000000000")
+            self.assertEqual(config.iot_submission_report, "PDF Informatieobjecttype")
+            self.assertEqual(config.iot_submission_csv, "CSV Informatieobjecttype")
+            self.assertEqual(config.iot_attachment, "Attachment Informatieobjecttype")
+
+        execute_single_step(
+            ObjectsAPIConfigurationStep, yaml_source=CONFIG_FILE_PATH_ALL_FIELDS
+        )
+
+        make_assertions()
+
+        execute_single_step(
+            ObjectsAPIConfigurationStep, yaml_source=CONFIG_FILE_PATH_ALL_FIELDS
+        )
+
+        make_assertions()
```

Comment:
I think the only relevant assertion for idempotency is that exactly one service gets created/updated, so I'd structure the test this way:

```py
assert not ObjectsAPIGroupConfig.objects.exists()

with self.subTest("run step first time"):
    execute_single_step(
        ObjectsAPIConfigurationStep, yaml_source=CONFIG_FILE_PATH_ALL_FIELDS
    )

    self.assertEqual(ObjectsAPIGroupConfig.objects.count(), 1)

with self.subTest("run step second time"):
    execute_single_step(
        ObjectsAPIConfigurationStep, yaml_source=CONFIG_FILE_PATH_ALL_FIELDS
    )

    # no additional configs created, but existing one updated
    self.assertEqual(ObjectsAPIGroupConfig.objects.count(), 1)
```

The assertions about what's inside the API group are not relevant for this test, you have a different test to check that behaviour. The idempotency test and assertions should only concern themselves with that idempotent bit.